controller/controller_client.go

```diff
@@ -23,266 +20,234 @@ type doneChan chan struct{}
 
 // A little helper for managing the state of the main runloop
 type clientHandler struct {
-	// tells the shipper server to begin shutdown
-	shutdownInit doneChan
-	// Tells the controller that the shipper backend has gracefully shut down.
-	shutdownComplete sync.WaitGroup
-
-	unitsMut sync.Mutex
-	units    map[string]*client.Unit
-
-	// the unit ID for the shipper itself
-	shipperOutputID string
+	// hashmap of units we get
+	units *UnitMap
 
-	shipperIsStopping uint32
-
-	log *logp.Logger
+	//shipperIsStopping uint32
+	shipperIsRunning bool
+	runner           *ServerRunner
+	log              *logp.Logger
 }
 
 func newClientHandler() clientHandler {
 	return clientHandler{
-		shutdownInit:      make(doneChan, 1),
-		log:               logp.L(),
-		units:             make(map[string]*client.Unit),
-		shipperIsStopping: 0,
+		log:              logp.L(),
+		units:            NewUnitMap(),
+		shipperIsRunning: false,
 	}
 }
 
-// Run starts the gRPC server
-func (c *clientHandler) Run(cfg config.ShipperConfig, unit *client.Unit) (err error) {
-	_ = unit.UpdateState(client.UnitStateConfiguring, "Initialising shipper server", nil)
-	runner, err := NewServerRunner(cfg)
-	if err != nil {
-		return err
+/*
+//////////
+////////// Shipper/output Handlers
+/////////
+*/
+
+// BounceShipper performs a (re)start of the shipper's backend components when triggered by another component
+func (c *clientHandler) BounceShipper() {
+	// TODO: once https://github.com/elastic/elastic-agent-shipper/issues/225 is done, a lot of this logic will need to change
+	if outUnit, gRPCConfig, ok := c.units.ShipperConfig(); ok { // do we have an output config?
+		if state, _, _ := outUnit.Expected(); state == client.UnitStateHealthy { // does the agent want us to be running?
+			// start shipper
+			if !c.shipperIsRunning {
+				c.log.Debugf("Starting shipper")
```

Comment:
This and the other logs here feels like they should be at the Info level. Changes shouldn't be common, and we want to know when it happens.